File: backend/app/schema/listing.py
from pydantic import BaseModel
from typing import List, Optional
from datetime import datetime

class VehicleListing(BaseModel):
    id: str
    seller_id: str
    make: str
    model: str
    year: int
    mileage: int
    price: float
    condition: str
    photos: List[str]
    maintenance_records: List[dict]
    status: str
    created_at: datetime
    updated_at: datetime

File: frontend/src/schema/listing.ts
import { z } from 'zod';

export const VehicleListingSchema = z.object({
  id: z.string(),
  sellerId: z.string(),
  make: z.string(),
  model: z.string(),
  year: z.number().int().positive(),
  mileage: z.number().nonnegative(),
  price: z.number().positive(),
  condition: z.string(),
  photos: z.array(z.string().url()),
  maintenanceRecords: z.array(z.string()),
  status: z.string(),
  createdAt: z.date(),
  updatedAt: z.date()
});

export type VehicleListing = z.infer<typeof VehicleListingSchema>;